File: descripe/imapp_modisl2/src/leocat/src/MOD_PRAlg29/set_confdnc.c
/*
************************************************************************
!F77 
 
!Description:
  Routine for setting output "bit" flags according
  to final confidence of clear sky.

!Input Parameters:
  None
  pxin.confdnc (final confidence of clear sky) is stored in pixel.h.

!Output Parameters:
  None
  pxout.testbits (6 byte array containing cloud mask bit flags) is
  stored in pixel.h.

!Revision History:
  R. Frey    10-25-07  Converted to C

!Team-Unique Header:

!References and Credits:

!END
***********************************************************************/

/* Includes */

#include <stdio.h>
#include <stdlib.h>
#include "pixel.h"
#include "mask_processing_constants.h"
      
void set_confdnc()
      
{      

   extern void set_bit(int, unsigned char[]);
   
	
   if(pxout.confidence > conf_clr_thresh) {
     (void) set_bit(1, pxout.testbits);
	 (void) set_bit(2, pxout.testbits);
   }
   else if(pxout.confidence > prob_clr_thresh) {
	 (void) set_bit(2, pxout.testbits);
   }	 
   else if(pxout.confidence > prob_cld_thresh) {
	 (void) set_bit(1, pxout.testbits);
   }


}

File: descripe/imapp_modisl2/src/cloudmask/src/set_confdnc.f
      subroutine set_confdnc(confdnc,testbits)

      implicit none
      save

c--------------------------------------------------------------------
C!F77 
c
c!Description:
c     Routine for setting output "bit" flags according
c     to final confidence of clear sky.
c
c!Input Parameters:
c confdnc       Unobstructed fov confidence
c
c!Output Parameters:
c testbits      6 cell byte array containing cloud mask bit results
c
c!Revision History:
c
c!Team-Unique Header:
c
c!References and Credits:
c See Cloud Mask ATBD-MOD-06.
c
c!END
c--------------------------------------------------------------------

c ... scalar arguments
      real confdnc

c ... array arguments
      byte testbits(6)

c ... local scalars
      integer debug,h_output

c ... Common statement for debug purposes
      common / bug / debug, h_output

c ... debug statement ............................................
      if (debug .gt. 0) then
        write(h_output,'(10x/,''Within set_confdnc routine '',/)')
      endif
c ...............................................................

      if(confdnc .gt. 0.99) then
        call set_bit(testbits,1)
        call set_bit(testbits,2)
      else if(confdnc .gt. 0.95) then
        call set_bit(testbits,2)
      else if(confdnc .gt. 0.66) then
        call set_bit(testbits,1)
      end if

c ... debug statement ............................................
      if (debug .gt. 0) then
        write(h_output,'(10x,'' confidence value: '',f10.2,/)')
     *        confdnc
        write(h_output,'(10x,'' First byte values: '',i10,/)')
     *        testbits(1)
      endif
c ................................................................

      return
      end
